src/GitHubVulnerabilities2Db/GraphQL/QueryService.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using GitHubVulnerabilities2Db.Configuration;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace GitHubVulnerabilities2Db.GraphQL
+{
+    public class QueryService : IQueryService
+    {
+        /// <remarks>
+        /// GitHub requires that every request includes a UserAgent.
+        /// </remarks>
+        public const string UserAgent = "NuGet.Jobs.GitHubVulnerabilities2Db";
+
+        public QueryService(
+            InitializationConfiguration initializationConfiguration,
+            HttpClient client)
+        {
+            InitializationConfiguration = initializationConfiguration ?? throw new ArgumentNullException(nameof(initializationConfiguration));
+            Client = client ?? throw new ArgumentNullException(nameof(client));
+        }
+
+        public InitializationConfiguration InitializationConfiguration { get; set; }
+        public HttpClient Client { get; set; }
+
+        public async Task<QueryResponse> QueryAsync(string query, CancellationToken token)
+        {
+            var queryJObject = new JObject
+            {
+                ["query"] = query
+            };
+
+            var response = await GetResponseFromQuery(queryJObject.ToString(), token);
+            return JsonConvert.DeserializeObject<QueryResponse>(response);
+        }
+
+        private async Task<string> GetResponseFromQuery(string query, CancellationToken token)
```

Comment:
```suggestion
        private async Task<string> GetResponseFromQueryAsync(string query, CancellationToken token)
```

Also, what do you think of the name `ExecuteQueryAsync`?

Reply:
This method is for specifically making the web request to the Graph QL API, I don't think `ExecuteQueryAsync` is appropriate here. How do you feel about `MakeWebRequestAsync`?

Reply:
I don't feel strongly about this, all names are fine by me 😊